lib/rules/no-duplicate-selectors/__tests__/index.js

```diff
@@ -84,6 +84,12 @@ testRule(rule, {
     message: messages.rejected("a b"),
     line: 1,
     column: 12,
+  }, {
+    code: "a { & {} }",
+    description: "duplicate caused by &-parent selector",
+    message: messages.rejected("a"),
```

Comment:
> we should just give any rule with just & as its selector a wide-berth i.e. don't resolve it and don't check it. If we go down this path, should we do it in stylelint or within postcss-resolve-nested-selector?

I think that for all intents and purposes `a { & { color: pink; } }` includes a duplicate selector, and over in #2198 everybody seems to agree about that. So it should be flagged.

However, because of #2198 and #2199 it looks like some people want an option to *not* consider the lone `&` to be a duplicate selector. That could be a new feature request.